src/NuGet.Server/Infrastructure/IServerPackageRepository.cs

```diff
@@ -1,12 +1,27 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
 
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Versioning;
+
 namespace NuGet.Server.Infrastructure
 {
     public interface IServerPackageRepository 
         : IServiceBasedRepository
     {
         void ClearCache();
+
         void RemovePackage(string packageId, SemanticVersion version);
+
+        IQueryable<ServerPackage> GetPackages(int semVerLevel);
+
+        IEnumerable<ServerPackage> FindPackagesById(string packageId, int semVerLevel);
```

Comment:
>semVerLevel [](start = 74, length = 11)

This implementation of support of SemVer 2 hardcodes semVer into a lot of interfaces. I would like to suggest a more general approach: Client compatibility filter. That will allow us in the future to specify additional parameters by which the client can pass the input he support easily.